File: mobile/lib/services/api/reports/reports_service.dart
import 'package:dio/dio.dart';
import 'package:injectable/injectable.dart';
import 'package:mobile/app_config.dart';
import 'package:mobile/models/models.dart';
import 'package:mobile/services/data_service.dart';

@injectable
class ReportsService extends DataService{
  ReportsService(AppConfig config) : super(config);

   Future<Report> addReport(Report report) async {
    Response response = await dio.post(apiUrl + "/reports", data: report);

    return Report.fromJson(response.data);
  }

}

File: webApp/src/app/services/api/reports/reports.service.ts
import { Report } from './../../../shared/models/dataModels';
import { DataService } from './../data.service';
import { Injectable } from '@angular/core';
import { catchError } from 'rxjs/internal/operators/catchError';

@Injectable({
  providedIn: 'root'
})
export class ReportsService extends DataService {

  addReport(report: Report) {
    return this.http.post<Report>(`${this.url}/reports`, report).pipe(catchError(this.handleError))
  }

}
